### PR TITLE
OCPBUGS-23115: Clarify instructions for builds with subscriptions

### PR DESCRIPTION
**Pull request by adambkaplan**
Builds that install RHEL packages need to include the `RUN rm -rf /etc/rhsm-host` instruction in the Dockerfile. This effectively tricks the subscription manager into thinking it is running outside of a container. Builds may also need to enable additional RHEL repositories if the package is not included in the base UBI or RHEL app stream repos.

This was tested with the latest UBI 7, 8, and 9 images as of 2024-01-26.

Version(s):
4.12 and higher

Issue:
[OCPBUGS-23115](https://issues.redhat.com/browse/OCPBUGS-23115)

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by adambkaplan** [comment]
@joaedwar please use the content here or add suggestions to fix up the asciidoc. Perhaps this replaces #68699?

**Comment by adambkaplan** [comment]
cc @gabemontero @jitendar-singh for doc clean up and QE verification.

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc [hunk 1]
```diff
@@ -1,197 +1,118 @@
...
   source:
     dockerfile: |
       FROM registry.redhat.io/ubi9/ubi:latest
-      RUN ls -la /etc/pki/entitlement
-      RUN rm /etc/rhsm-host
-      RUN yum repolist --disablerepo=*
-      RUN subscription-manager repos --enable rhocp-4.9-for-rhel-8-x86_64-rpms
-      RUN yum -y update
-      RUN yum install -y openshift-clients.x86_64
+      RUN rm -rf /etc/rhsm-host <1>
```
[the remark this answers is not in the document]
Comment (adambkaplan) [reply modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc:85]: Thanks for pointing out the example in the shared resource CSI driver repo - I'll update the example so we're all aligned.

Indeed cluster-bot testing is not feasible as CI clusters are typically not subscribed. https://github.com/openshift/origin/pull/28560 has logic to skip its tests if the entitlement keys aren't found. I was able to run the tests manually on ROSA clusters (which are subscribed and have entitlements, including for hosted control plane). Shared resources isn't in that suite yet - I think we should place that in our eventual builds for OpenShift layered product e2e tests.

**Review by nebelsprachen — APPROVED** [review]
looks good. Just one tiny nit :)

File: modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc [hunk 2]
```diff
@@ -1,197 +1,118 @@
...
 [id="builds-running-entitled-builds-with-sharedsecret-objects_{context}"]
-= Running entitled builds using SharedSecret objects
+= Running builds using SharedSecret objects
 
-You can configure and perform a build in one namespace that securely uses RHEL entitlements from a `Secret` object in another namespace.
+Starting in {product-title} 4.10 and later, you can use a `SharedSecret` object to securely access the cluster's entitlement keys in builds.
+`SharedSecret` objects allow `Secret` content to be shared across namespaces and kept synchronized.
 
-You can still access RHEL entitlements from OpenShift Builds by creating a `Secret` object with your subscription credentials in the same namespace as your `Build` object. However, now, in {product-title} 4.10 and later, you can access your credentials and certificates from a `Secret` object in one of the {product-title} system namespaces. You run entitled builds with a CSI volume mount of a `SharedSecret` custom resource (CR) instance that references the `Secret` object.
+This procedure relies on the Shared Resource CSI Driver feature.
```
[the remark this answers is not in the document]
Comment (adambkaplan) [reply modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc]: @joaedwar suggestion accepted!

**Comment by adambkaplan** [comment]
@joaedwar bump again - is this :+1: to be merged?

**Pull request closed without merging**